v2/tools/generator/internal/astmodel/enum_type.go

```diff
@@ -40,9 +41,28 @@ func NewEnumType(baseType *PrimitiveType, options ...EnumValue) *EnumType {
 		panic("baseType must be provided")
 	}
 
-	sort.Slice(options, func(left int, right int) bool {
-		return options[left].Identifier < options[right].Identifier
-	})
+	slices.SortFunc(
+		options,
+		func(left EnumValue, right EnumValue) int {
+			if baseType == IntType {
+				// Compare integers as numbers if we can
+				l, lOk := strconv.Atoi(left.Identifier)
```

Comment:
Is this also duplicate from your other PR?

Reply:
Hmm maybe not?

Reply:
I pulled it out of here for a different PR, knowing I'd need to rebase this one.